### PR TITLE
CFINSPEC-402: k8s resources Lint, unit test and doc fixes

### PR DESCRIPTION
**Pull request by Vasu1105**
## Description

This PR 
* fixes a few lint failures ( The easy to fix one. There are some code complexity and other lint failures but I am not addressing them in this PR. Those need detail discussion)
* fixes test for labels and annotations for `k8s_config_map`, `k8s_config_maps`, k8s_pod, k8s_pods resources
* Minor doc fixes - Removed type property for` k8s_config_map` and `k8s_config_maps` resource
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: libraries/k8sversion.rb [hunk 1]
```diff
@@ -14,7 +15,7 @@ class K8sVersion < K8sResourceBase
         ...
       end
     "
-    attr_reader :k8sversion, :k8suri, :gitVersion, :major, :minor, :platform, :buildDate
+    attr_reader :k8sversion, :k8suri, :gitVersion, :major, :minor, :buildDate
```
Comment (Vasu1105) [inline libraries/k8sversion.rb:18]: We have a method platform in the same class, so lint gave a duplicate method error.

File: libraries/k8sobjects.rb [hunk 2]
```diff
@@ -46,7 +47,7 @@ def initialize(opts = {})
         @objtype = opts[:type] if opts[:type] ||= nil
         @objname = opts[:name] if opts[:name] ||= nil
         @objnamespace = opts[:namespace] if opts[:namespace] ||= nil
-        @objlabelSelector = opts[:labelSelector] if opts[:labelSelector] ||= nil
+        @obj_label_selector = opts[:labelSelector] if opts[:labelSelector] ||= nil
```
Comment (Vasu1105) [inline libraries/k8sobjects.rb:50]: camel case fix

**Review by ahasunos — COMMENTED** [review]
Does the matcher link gets referenced from some place?
```
## Matchers

{{% inspec/inspec_matchers_link %}}
```

**Review by ahasunos — APPROVED** [review]

**Review by Nik08 — APPROVED** [review]
Looks good to me since these are just lint changes!

**Pull request merged**